vernac/comAssumption.ml

```diff
@@ -58,7 +40,11 @@ let instance_of_univ_entry = function
   | Monomorphic_entry _ -> Univ.Instance.empty
 
 let declare_axiom is_coe ~poly ~local ~kind typ (univs, pl) imps nl {CAst.v=name} =
-  let do_instance = should_axiom_into_instance kind in
+  let do_instance = let open Decls in match kind with
+  | Context -> true
+    (* The typeclass behaviour of Variable and Context doesn't depend
+       on section status *)
+  | Definitional | Logical | Conjectural -> false in
```

Comment:
Maye it's just me but I like the `in` to be on a line on its own when it can't be on the same line as `let`

Reply:
I like this too, I don't remember why I did it that way, maybe to look like the surroundings.